src/Process/3dTilesProcessing.js

```diff
@@ -88,6 +100,51 @@ export function pre3dTilesUpdate(context, layer) {
     // const HFOV = 2.0 * Math.atan(Math.tan(radAngle * 0.5) / context.camera.ratio);
     const HYFOV = 2.0 * Math.atan(Math.tan(radAngle * 0.5) * hypotenuse / context.camera.width);
     context.camera.preSSE = hypotenuse * (2.0 * Math.tan(HYFOV * 0.5));
+
+    // once in a while, garbage collect
+    if (Math.random() < 0.98) {
+        const now = Date.now();
+        layer.root.traverse((n) => {
+            if (n.layer != layer.id) {
+                return;
+            }
+
+            // Browse children, and remove 'deletable' ones before
+            // 'traverse' method visit them.
+            for (let i = 0; i < n.children.length; i++) {
+                const c = n.children[i];
+                if (c.layer != layer.id) {
+                    continue;
+                }
+
+                if ((now - c.notVisibleSince) > (layer.cleanupDelay || 3000)) {
+                    c.traverse((o) => {
+                        // Note: we don't check o.layer since we're
+                        // going to remove 'c' from its parent so let's
+                        // clean all of its children.
+
+                        // free resources
+                        if (o.material) {
+                            o.material.dispose();
+                        }
+                        if (o.geometry) {
+                            o.geometry.dispose();
+                        }
+                        // we can't remove 'o' from 'c' yet,
+                        // because 'traverse' first applies the callback to a node
+                        // and then to its children.
+                    });
+
+                    // remove c children recursively
+                    removeChildren(c);
+                    delete c.content;
```

Comment:
AFAIK `c.content` can actually be a `Scene` object (when it is a parsed gltf for instance), and this scene contains *another* hierarchy of 3D objects you need to clean...

Reply:
You want to add this logic to `removeChildren` too I guess

Reply:
(I refactored this part a bit)
But even with the current version, the full hierarchy should be traversed for `dispose()` and then removed.